packages/devtools_app/lib/src/history_viewport.dart

```diff
@@ -0,0 +1,90 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import 'common_widgets.dart';
+import 'debugger/common.dart';
+import 'history_manager.dart';
+import 'theme.dart';
+
+/// A [Widget] that allows for displaying content based on the current state of a
+/// [HistoryManager]. Includes built-in controls for navigating back and forth
+/// through the content stored in the provided [HistoryManager].
+///
+/// [history] is the [HistoryManger] that contains the data to be displayed.
+///
+/// [buildContents] is invoked with the currently selected historical data
+/// when building the contents of the viewport.
+///
+/// If [buildControls] is provided, the returned [List<Widget>] will be
+/// inserted at the end of the viewport title bar.
+///
+/// If [generateTitle] is provided, the title string will be set to the
+/// returned value. If not provided, the title will be empty.
+class HistoryViewport<T> extends StatelessWidget {
+  const HistoryViewport({
```

Comment:
we should be able to make this class more declarative instead of using builders. Instead of using builders and generators, just pass in `String title, List<Widget> controls, Widget child`. Sound like the problem is that there is some state that is not getting updated when the selected script changes? If that's the case, then I think the proper way to make sure this widget rebuilds is to wrap it in a ValueListenable builder where this widget is used, or to make this widget stateful and add listeners to appropriate notifiers in initState